Make format and dhall subcommands of dhall executable

gilligan opened the pull request:
I really just moved code around a bit - fixes #436 

I just realized that Tutorial.hs also has to be adjusted since it is using `dhall-format` and `dhall-hash`. Another thing is with `--inplace` now being a global switch. Is that Ok @Gabriel439 or should that rather be moved down to be a switch to the `format` subcommand?

gilligan commented:
Oops, just saw the upstream changes and fixing..

Gabriel439 commented on src/Dhall/Main.hs:
```diff
@@ -65,6 +68,13 @@ parseOptions = Options <$> parseMode <*> parseExplain <*> parsePlain
             <>  Options.Applicative.help "Disable syntax highlighting"
             )
 
+    parseInplace =
```
Yeah, I think this should be a `format`-specific option (See https://github.com/dhall-lang/dhall-haskell/pull/451 for an example of how to do that)

Note that it's still worth keeping the intermediate `parseInplace` utility as I imagine that other utilities might reuse it in the future (like the `dhall freeze` command described in https://github.com/dhall-lang/dhall-haskell/issues/437)

gilligan replied:
:ok_hand:

gilligan commented:
@Gabriel439 i also updated Tutorial.hs, let me know if that works for you.

@fingerzam I suppose your PR was actually ready before mine. I somehow missed it. Sorry. I actually took the format subcommand code from you now but I moved hash and format to their own modules..

Gabriel439 commented:
As I mentioned in #436 I have decided to merge this pull request.  Thanks to both of you, though, for taking the time to do this 🙂
